### PR TITLE
Employee controller

### PR DESCRIPTION
**Pull request by Apalok-Kreeti**
Created Employee Controller & View

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -3,8 +3,6 @@ source 'https://rubygems.org'
 gem 'rails', '~> 5.0.0', '>= 5.0.0.1'
 gem 'puma', '~> 3.0'
 gem 'coffee-rails', '~> 4.2'
-gem 'devise'
-gem 'devise-bootstrap-views'
```
Comment (nbanerjee) [inline Gemfile]: Why devise related changes are showing here?

Comment (nbanerjee) [inline Gemfile]: Plz rebase your branch

File: app/assets/javascripts/employees.coffee [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
+# All this logic will automatically be available in application.js.
+# You can use CoffeeScript in this file: http://coffeescript.org/
```
Comment (nbanerjee) [inline app/assets/javascripts/employees.coffee]: Remove unwanted files

File: app/assets/stylesheets/employees.scss [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    text-align: left;
+}
+
+th, td {
+    padding: 15px;
+}
+
+table {
+    border-spacing: 5px;
+    width: 100%;
```
Comment (nbanerjee) [inline app/assets/stylesheets/employees.scss]: Indent codes and write proper css.

File: app/controllers/employees_controller.rb [hunk 4]
```diff
@@ -0,0 +1,55 @@
+class EmployeesController < ApplicationController
+
```
Comment (nbanerjee) [inline app/controllers/employees_controller.rb]: Remove unwanted newlines.

File: app/controllers/employees_controller.rb [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+  end
+
+  def index
+    @employees = Employee.paginate(page: params[:page], per_page:10)
+  end
+
+  def destroy
+    Employee.find(params[:id]).destroy
+    flash[:success] = "Employee Details Deleted"
+    redirect_to employees_path
```
Comment (nbanerjee) [inline app/controllers/employees_controller.rb]: What happen if some error occurred at delete?

File: app/controllers/employees_controller.rb [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+  end
+
+  def show
+    @employee = Employee.find(params[:id])
+  end
```
Comment (nbanerjee) [inline app/controllers/employees_controller.rb]: Plz follow DRY, cleanup using before action.

File: test/controllers/employees_controller_test.rb [hunk 7]
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
+
+class EmployeesControllerTest < ActionDispatch::IntegrationTest
+  # test "the truth" do
+  #   assert true
+  # end
+end
```
Comment (nbanerjee) [inline test/controllers/employees_controller_test.rb]: Remove unwanted files, related this PR

File: app/helpers/employees_helper.rb [hunk 8]
```diff
@@ -0,0 +1,2 @@
+module EmployeesHelper
+end
```
Comment (nbanerjee) [inline app/helpers/employees_helper.rb]: Remove unwanted files.

File: app/views/employees/index.html.erb [hunk 9]
```diff
@@ -0,0 +1,32 @@
+<% provide(:title, 'All Empoyees Details') %>
+<%= link_to 'Back', root_path, class: "btn btn-lg btn-primary", id: "button1" %>
+<br><br>
```
Comment (nbanerjee) [inline app/views/employees/index.html.erb]: Remove br and use css

File: app/views/employees/new.html.erb [hunk 10]
```diff
@@ -0,0 +1,30 @@
+<% provide(:title, 'Add Employee Details') %>
+<%= link_to 'Back', employees_path, class: "btn btn-lg btn-primary", id: "button1" %>
+<br>
+<h1 class="center">New Employee</h1>
+
+<div class="row">
+  <div class="col-md-6 col-md-offset-3">
+    <%= form_for(@employee, url: employees_path) do |f| %>
+      <%= render 'shared/error_employees', obj: @employee %>
+      <br>
```
Comment (nbanerjee) [inline app/views/employees/new.html.erb]: Remove all br and use bootstrap.

**Comment by Apalok-Kreeti** [comment]
Deleted unwanted files, used before action, removed line space

File: app/controllers/employees_controller.rb [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+  private
+
+  def get_empID
+    @employee = Employee.find(params[:id])
+  end
+
+  def emp_params
+    params.require(:employee).permit(:name, :email, :mobile, :designation, :active)
+  end
+
```
Comment (nbanerjee) [inline app/controllers/employees_controller.rb]: Remove newline

File: app/controllers/employees_controller.rb [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+  def update
+
+    if @employee.update_attributes(emp_params)
+      flash[:success] = "Employee Details Updated"
+      redirect_to employee_path(@employee)
+    else
+      render 'edit'
+    end
+
```
Comment (nbanerjee) [inline app/controllers/employees_controller.rb]: Remove newline

File: app/controllers/employees_controller.rb [hunk 13]
```diff
@@ -0,0 +1,55 @@
+class EmployeesController < ApplicationController
+  before_action :get_empID, only: [:edit, :update, :show, :destroy]
+
```
Comment (nbanerjee) [inline app/controllers/employees_controller.rb]: Change method to get_employee.

File: app/views/employees/edit.html.erb [hunk 14]
```diff
@@ -0,0 +1,38 @@
...
+      </table>
+      <br>
+      <div class="center">
+        <%= f.submit "Update Details", class: "btn btn-primary" %>
+      </div>
+    <% end %>
+  </div>
+</div>
+
+<br>
```
Comment (nbanerjee) [inline app/views/employees/edit.html.erb]: Remove all br and use css & bootstrap.

File: app/views/shared/_error_employees.html.erb [hunk 15]
```diff
@@ -0,0 +1,17 @@
...
+        <%= pluralize(obj.errors.count, "error") %>
+        prohibited this details from being saved:
+      </h2>
+      <div class="panel-body">
+        <ul>
+          <% obj.errors.full_messages.each do |msg| %>
+            <li><%= msg %></li>
+          <% end %>
+        </ul>
+      </div>
```
Comment (nbanerjee) [inline app/views/shared/_error_employees.html.erb]: Why it's only for employee? Make it generic and use.

**Comment by Apalok-Kreeti** [comment]
deleted buffer files & used css

**Comment by Apalok-Kreeti** [comment]
modified

File: app/controllers/employees_controller.rb [hunk 16]
```diff
@@ -0,0 +1,52 @@
...
+    if @employee.save
+      flash[:success] = "Employee with #{@employee.name} is Created Successfully!"
+      redirect_to employee_path(@employee)
+    else
+      render 'new'
+    end
+  end
+
+  def update
+
```
Comment (nbanerjee) [inline app/controllers/employees_controller.rb]: Plz remove all of this newlines

File: app/views/employees/index.html.erb [hunk 17]
```diff
@@ -0,0 +1,33 @@
...
+      <td> <%=link_to 'Edit', edit_employee_path(employee), class: 'btn btn-default' %> </td>
+      <td> <%=link_to 'Delete', employee_path(employee), method: :delete, data: {confirm: 'Do you want to delete the Information?'}, class: 'btn btn-default' %> </td>
+    </tr>
+  <% end %>
+</table>
+
+<%= will_paginate %>
+<div class="form-group">
+  <%=link_to 'Add New Employee', new_employee_path %>
+</div>
```
Comment (nbanerjee) [inline app/views/employees/index.html.erb]: Add new employee should be top right, before that hr.

**Comment by Apalok-Kreeti** [comment]
plz go through

File: app/controllers/employees_controller.rb [hunk 18]
```diff
@@ -0,0 +1,51 @@
...
+    end
+  end
+
+  private
+
+  def get_employee
+    @employee = Employee.find(params[:id])
+  end
+
+  def emp_params
```
Comment (nbanerjee) [inline app/controllers/employees_controller.rb]: It should be employee_params

File: app/views/employees/edit.html.erb [hunk 19]
```diff
@@ -0,0 +1,40 @@
...
+</div>
+<h1 class="center">Edit Details</h1>
+
+<div class="row">
+  <div class="col-md-6 col-md-offset-3">
+    <%= form_for(@employee) do |f| %>
+      <%= render 'shared/errors', obj: @employee %>
+      <div class="form-group">
+      <%= f.label :name , 'Name of the Employee:' %>
+      <%= f.text_field :name, class: 'form-control' %>
```
Comment (nbanerjee) [inline app/views/employees/edit.html.erb]: Fix all indentations.

File: config/application.rb [hunk 20]
```diff
@@ -11,5 +11,6 @@ class Application < Rails::Application
     # Settings in config/environments/* take precedence over those specified here.
     # Application configuration should go into files in config/initializers
     # -- all .rb files in that directory are automatically loaded.
+    WillPaginate.per_page = 2
   end
```
Comment (nbanerjee) [inline config/application.rb]: Make it 20

File: app/assets/stylesheets/employees.scss [hunk 21]
```diff
@@ -0,0 +1,40 @@
...
+// You can use Sass (SCSS) here: http://sass-lang.com/
+
+//universal
+
+.center {
+  text-align: center;
+}
+
+// typography
+
```
Comment (nbanerjee) [inline app/assets/stylesheets/employees.scss]: Remove unwanted comments

**Pull request merged**